### DISCUSSION TITLE
Can't get LSPs to run

### DISCUSSION THREAD

**Discussion by tonekk** [NvChad/NvChad#1462] [Q&A] [answered]
Hi,

I followed the guide on the homepage, even copied the exact files from @siduck 's dotfiles, but I can't get any other lsps to run.
Installed them all using `MasonInstallAll`, but `LspInfo` only shows lua.
When I add them in `lspconfig` in the `plugins` directory, it works.

I'm struggling to debug what is happening (I guess because we utilize `pcall` to load `chadrc`. What could be the problem?

Here are my dotfiles:
https://github.com/tonekk/nvchad-config

**Comment by siduck** [comment, marked as answer]
@tonekk run packercompile

**Reply by tonekk** [reply]
Holy shit m8, you're the GOAT :)
Can you briefly explain why that is necessary?

**Reply by siduck** [reply]
# Plugin Definition ( happens in default & custom plugin list ) 

![image](https://user-images.githubusercontent.com/59060246/184166286-ead2722e-1c6f-42e8-822e-ca238c029149.png)

- This screenshot is what we call a plugin definition and the arrows point towards the packer options. 

- There are tons of packer options, as per packer readme 

![image](https://user-images.githubusercontent.com/59060246/184166593-1db43197-00ef-4e0f-9289-0c39f2f9c491.png)

# How NvChad manages custom plugins? 

![image](https://user-images.githubusercontent.com/59060246/184169455-1f6807c2-564c-49d2-997c-fc644a9cc9d4.png)

# Why does packercompile work?

 ```lua
["neovim/nvim-lspconfig"] = {
    config = function()
      require "plugins.configs.lspconfig"
      require "custom.plugins.lspconfig"
    end,
},
```

- Ok so when you add the above in custom plugins, it changes the `config` key in the default plugin list's `'neovim/nvim-lspconfig'` key and a `packercompile` is needed to save any changes that's related to packer i.e your custom plugins config basically merges into default plugins config and the default plugins config is related to wrapped into a packer function.

- And Doing `packercompile` updates the old `packer_compiled.lua` file which is in `.config/nvim/lua/plugin` folder

**Reply by tonekk** [reply]
Awesome explanation, thanks 👍
